### DISCUSSION TITLE
How to exclude a service from AutoconfigureTag on base class or interface?

### DISCUSSION THREAD

**Discussion by donquixote** [symfony/symfony#57196] [Q&A]
I have an interface `MyInterface` in package A with `#[AutoconfigureTag('mytag')]`.
I have a specific class `MyClass` in package B that implements `MyInterface`.

I want this class to be registered as a service, but I don't want it to be tagged.
How would I achieve this?

For the sake of this question, assume that I only control package B, but not package A.

Perhaps the same solution will work for any kind of autoconfigure. But for now the question is about tags.

**Comment by MatTheCat** [comment]
The only way I can think of is removing the tag with a [compiler pass](https://symfony.com/doc/current/service_container/compiler_passes.html):

```php
$container->getDefinition(MyClass::class)->clearTag('mytag');
```

**Reply by donquixote** [reply]
Alright, but not ideal.
Maybe a better solution would be a custom attribute plus `->registerAttributeForAutoconfiguration()`.

**Reply by MatTheCat** [reply]
You can open a [feature request](https://github.com/symfony/symfony/issues/new?template=2_Feature_request.yaml); I believe this would bring more attention.
